cfgmgr/buffermgrdyn.cpp

```diff
@@ -0,0 +1,1620 @@
+                    if (fvField(fv) == "gearbox_model")
+                        gearboxModel = fvValue(fv);
+                }
+            }
+        }
+
+        m_identifyGearboxDelay = m_gearboxDelay[gearboxModel];
+        m_supportGearbox = false;
+    }
+}
+
+void BufferMgrDynamic::initTableHandlerMap()
+{
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(STATE_BUFFER_MAXIMUM_VALUE_TABLE, &BufferMgrDynamic::handleBufferMaxParam));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_DEFAULT_LOSSLESS_BUFFER_PARAMETER, &BufferMgrDynamic::handleDefaultLossLessBufferParam));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_POOL_TABLE_NAME, &BufferMgrDynamic::handleBufferPoolTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_PROFILE_TABLE_NAME, &BufferMgrDynamic::handleBufferProfileTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_QUEUE_TABLE_NAME, &BufferMgrDynamic::handleBufferQueueTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_PG_TABLE_NAME, &BufferMgrDynamic::handleBufferPgTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_PORT_INGRESS_PROFILE_LIST_NAME, &BufferMgrDynamic::handleBufferPortIngressProfileListTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_BUFFER_PORT_EGRESS_PROFILE_LIST_NAME, &BufferMgrDynamic::handleBufferPortEgressProfileListTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_PORT_TABLE_NAME, &BufferMgrDynamic::handlePortTable));
+    m_bufferTableHandlerMap.insert(buffer_handler_pair(CFG_PORT_CABLE_LEN_TABLE_NAME, &BufferMgrDynamic::handleCableLenTable));
+}
+
+// APIs to handle variant kinds of keys
+
+// Transform key from CONFIG_DB format to APPL_DB format
+void BufferMgrDynamic::transformSeperator(string &name)
+{
+    size_t pos;
+    while ((pos = name.find("|")) != string::npos)
+        name.replace(pos, 1, ":");
+}
+
+void BufferMgrDynamic::transformReference(string &name)
+{
+    auto references = tokenize(name, list_item_delimiter);
+    int ref_index = 0;
+
+    name = "";
+
+    for (auto &reference : references)
+    {
+        if (ref_index != 0)
+            name += list_item_delimiter;
+        ref_index ++;
+
+        auto keys = tokenize(reference, config_db_key_delimiter);
+        int key_index = 0;
+        for (auto &key : keys)
+        {
+            if (key_index == 0)
+                name += key + "_TABLE";
+            else
+                name += delimiter + key;
+            key_index ++;
+        }
+    }
+}
+
+// For string "TABLE_NAME|objectname", returns "objectname"
+string BufferMgrDynamic::parseObjectNameFromKey(const string &key, size_t pos = 0)
+{
+    auto keys = tokenize(key, delimiter);
+    if (pos >= keys.size())
+    {
+        SWSS_LOG_ERROR("Failed to fetch %lu-th sector of key %s", pos, key.c_str());
```

Comment:
return some null string and the caller to check for these errors and handle the errors??

Reply:
Will fix it